spec/forms/state_file/nj_veterans_exemption_form_spec.rb

```diff
@@ -0,0 +1,53 @@
+require "rails_helper"
+
+RSpec.describe StateFile::NjVeteransExemptionForm do
+  describe "validations" do
+    context "when filing status is single and fields are empty" do
+      let(:intake) {
+        create :state_file_nj_intake, primary_veteran: nil, spouse_veteran: nil
+      }
+      let(:form) { described_class.new(intake) }
+      it "flags the primary_veteran field as an error but not the spouse_veteran field" do
+        expect(form.valid?).to eq false
+        expect(form.errors[:primary_veteran]).to include "Can't be blank."
+        expect(form.errors[:spouse_veteran]).to eq []
+      end
+    end
+
+    context "when filing status is married filing jointly and fields are empty" do
+      let(:intake) {
+        create :state_file_nj_intake, :married_filing_jointly, primary_veteran: nil, spouse_veteran: nil
+      }
+      let(:form) { described_class.new(intake) }
```

Comment:
[sand] If you wanted to reduce some duplication, you could use the `invalid_params` pattern like you do below and like we do in other form specs to avoid having to redeclare the `let form` and `let intake` in each situation